pkg/deploy/registry/deploylog/rest.go

```diff
@@ -0,0 +1,181 @@
+	return c.WatchRcFn(ctx, label, field, rv)
+}
+
+type podGetter struct {
+	podsNamespacer kclient.PodsNamespacer
+}
+
+// Get is responsible for retrieving the deployer pod
+func (g *podGetter) Get(ctx kapi.Context, name string) (runtime.Object, error) {
+	ns, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.NewBadRequest("namespace parameter required.")
+	}
+	return g.podsNamespacer.Pods(ns).Get(name)
+}
+
+const defaultTimeout time.Duration = 10 * time.Second
+
+// NewREST creates a new REST for DeployLogs. It accepts a DeploymentClient and a pod
+// client to get necessary attributes to assemble the URL to which the request shall be
+// redirected in order to get the deployment logs.
+func NewREST(d DeploymentClient, pn kclient.PodsNamespacer, connectionInfo kclient.ConnectionInfoGetter) *REST {
+	return &REST{
+		Deploy:         d,
+		PodGetter:      &podGetter{pn},
+		ConnectionInfo: connectionInfo,
+		Timeout:        defaultTimeout,
+	}
+}
+
+// REST implements GetterWithOptions
+var _ = rest.GetterWithOptions(&REST{})
+
+// Get returns a streamer resource with the contents of the deployment log
+func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.Object, error) {
+	deployLogOpts, ok := opts.(*deployapi.DeploymentLogOptions)
+	if !ok {
+		return nil, errors.NewBadRequest("did not get an expected options.")
+	}
+
+	config, err := r.Deploy.GetDeploymentConfig(ctx, name)
+	if err != nil {
+		return nil, errors.NewNotFound("deploymentConfig", name)
+	}
+
+	deployment, err := r.Deploy.GetDeployment(ctx, deployutil.LatestDeploymentNameForConfig(config))
+	if err != nil {
+		// TODO: Fallback to latest-1?
+		return nil, errors.NewNotFound("deployment", deployutil.LatestDeploymentNameForConfig(config))
+	}
+
+	status := deployutil.DeploymentStatusFor(deployment)
+	switch status {
+	case deployapi.DeploymentStatusNew, deployapi.DeploymentStatusPending:
+		if deployLogOpts.NoWait {
+			glog.V(4).Infof("Deployment %s/%s is in %s state. No logs to retrieve yet.", deployment.Namespace, deployment.Name, status)
+			return &genericrest.LocationStreamer{}, nil
+		}
+		glog.V(4).Infof("Deployment %s/%s is in %s state, waiting for it to start...", deployment.Namespace, deployment.Name, status)
+		if err := r.waitForDeployment(ctx, deployment); err != nil {
+			return nil, err
+		}
+	}
+
+	deployPodName := deployutil.DeployerPodNameForDeployment(deployment.Name)
+	logOpts := &kapi.PodLogOptions{
+		Follow: deployLogOpts.Follow,
+	}
+	location, transport, err := pod.LogLocation(r.PodGetter, r.ConnectionInfo, ctx, deployPodName, logOpts)
+	if err != nil {
+		return nil, errors.NewBadRequest(err.Error())
+	}
+	return &genericrest.LocationStreamer{
+		Location:    location,
+		Transport:   transport,
+		ContentType: "text/plain",
+		Flush:       deployLogOpts.Follow,
+	}, nil
+}
+
+func (r *REST) waitForDeployment(ctx kapi.Context, deployment *kapi.ReplicationController) error {
```

Comment:
I cheated because you asked for a reflector but I used what I'm more familiar with ie. informers:) I will have a look at reflectors shortly. Also, as discussed on IRC, I am going to extract all the common code into a library so build code can use it too.